### PR TITLE
Add support for figsize and masking to confusion matrix hook

### PR DESCRIPTION
**Pull request by FloopCZ**

### CODE REVIEW

File: cxflow/hooks/save_cm.py [hunk 1]
```diff
@@ -116,6 +119,7 @@ def after_epoch(self, epoch_id: int, epoch_data: EpochData) -> None:
             cm = cm_norm if self._normalize else cm_abs
 
             # Save the heatmap of confusion matrix
+            plt.figure(figsize=self._figsize)
```
Comment (blazekadam) [inline cxflow/hooks/save_cm.py]: Does this work with `figsize==None`?

Comment (FloopCZ) [reply cxflow/hooks/save_cm.py]: yes

**Comment by FloopCZ** [comment]
@petrbel Thank you for the review, the issue has been addressed.

**Comment by blazekadam** [comment]
@FloopCZ ouch... being denied by @coveralls guy...

File: cxflow/hooks/save_cm.py [hunk 2]
```diff
@@ -77,19 +81,31 @@ def __init__(self,
...
+        super().__init__(variables=accum_variables, **kwargs)
 
     def after_epoch(self, epoch_id: int, epoch_data: EpochData) -> None:
         for stream_name, variables in self._accumulator.items():
             predicted = np.array(variables[self._predictions_name])
             expected = np.array(variables[self._labels_name])
 
+            # Only use the masked data if requested
+            if self._mask_name is not None:
+                assert self._mask_name in variables
```
Comment (blazekadam) [inline cxflow/hooks/save_cm.py]: whoa... you don't even trust yourself :D :D :D

File: cxflow/tests/hooks/save_cm_test.py [hunk 3]
```diff
@@ -80,6 +81,14 @@ def test_after_epoch(self):
                                    figure_action='store')
         epoch_data = SaveConfusionMatrixTest.run_hook(hook)
         self.assertTupleEqual(tuple(epoch_data['train']['confusion_heatmap'].shape), (480, 640, 3))
+
+        # test changing figure size
+        hook = SaveConfusionMatrix(dataset=TestDataset(), output_dir='', figure_action='store',
+                                   figsize=(10, 15))
+        epoch_data = SaveConfusionMatrixTest.run_hook(hook)
+        dpi = matplotlib.rcParams['figure.dpi']
```
Comment (blazekadam) [inline cxflow/tests/hooks/save_cm_test.py]: :+1:

File: cxflow/hooks/save_cm.py (hunk 2 above)
Comment (FloopCZ) [reply cxflow/hooks/save_cm.py]: This is unnecessary, I will remove the assertion.

File: cxflow/tests/hooks/save_cm_test.py (hunk 3 above)
Comment (FloopCZ) [reply cxflow/tests/hooks/save_cm_test.py]: > @FloopCZ ouch... being denied by @coveralls guy...

Yeah, it is kind of hard to test that the picture is valid, I am unsure I will be able to do something about it.

**Comment by FloopCZ** [comment]
> @FloopCZ ouch... being denied by @coveralls guy...

Yeah, it is kind of hard to test that the picture is valid, I am unsure I will be able to do something about it.

**Comment by blazekadam** [comment]
Yea... I wanted a testable `compute_cm` function and harder to test plotting function but here we are...

So at least, you can test if the hook works with some mask variable? That would cover the lines (and make @coveralls guy happy) but also check if the syntax and types are valid...

File: cxflow/hooks/save_cm.py (hunk 2 above)
Comment (blazekadam) [reply cxflow/hooks/save_cm.py]: Yep `AccumulateVariables` hook is checking this already but you know.. there are people who prefer doubling the protection right? :)

**Comment by FloopCZ** [comment]
Good idea, will do, thanks. :+1:

**Comment by FloopCZ** [comment]
The issues have been addressed, thank you for your reviews.

**Pull request merged**